src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -8,20 +8,40 @@ import loadLanguages from './loadLanguages'
 import loadCategories from './loadCategories'
 import loadEvents from './loadEvents'
 import fetchResourceCache from './fetchResourceCache'
+import moment from 'moment-timezone'
 
-export default function * loadCityContent (database: MemoryDatabase, newCity: string, newLanguage: string): Saga<void> {
-  if (database.hasContext(new MemoryDatabaseContext(newCity, newLanguage))) {
-    return // All data is already in the database
+const TWENTY_FOUR_HOURS = 24
+
+export default function * loadCityContent (
+  database: MemoryDatabase, newCity: string, newLanguage: string, forceRefresh: boolean): Saga<void> {
+  if (database.hasContext(newCity, newLanguage)) {
+    return
   }
 
   database.changeContext(new MemoryDatabaseContext(newCity, newLanguage))
 
+  yield call(database.readLastUpdate)
+
+  console.debug('Last city content update on ',
+    database.lastUpdate ? database.lastUpdate.toISOString() : database.lastUpdate)
+
+  // The last update was more than 24h ago or a refresh should be forced
+  const shouldUpdate = forceRefresh || !database.lastUpdate ||
+    database.lastUpdate.isBefore(moment.tz('UTC').subtract(TWENTY_FOUR_HOURS, 'hours'))
+
+  console.debug('City content should be refreshed: ', shouldUpdate)
+
   const [categoryUrls, eventUrls] = yield all([
-    call(loadCategories, newCity, newLanguage, database),
-    call(loadEvents, newCity, newLanguage, database),
-    call(loadLanguages, newCity, database)
+    call(loadCategories, newCity, newLanguage, database, shouldUpdate),
+    call(loadEvents, newCity, newLanguage, database, shouldUpdate),
+    call(loadLanguages, newCity, database, shouldUpdate)
   ])
 
   const fetchMap = {...categoryUrls, ...eventUrls}
   yield call(fetchResourceCache, newCity, newLanguage, fetchMap, database)
+
+  if (shouldUpdate) {
+    database.lastUpdate = moment.tz('UTC')
+    yield call(database.writeLastUpdate)
```

Comment:
Can't we call `writeLastUpdate` automatically if some content in the memoryDatabase is updated?

Reply:
We completely remodeled the MemoryDatabase and defined a proper interface in #40. This will be updated and adjusted to the new interface, which does indeed automatically write.